R/read_file.R

```diff
@@ -29,12 +29,21 @@ read_file <- function(path, col_select = NULL, ...) {
     ))
   }
 
+  if (!(file.exists(path))) {
+    cli::cli_abort(
+      c(
+        "x" = "File {.val {fs::path_file(fs::path_ext_remove(path))}}
+             is NOT available",
+        "i" = "Contact phs.geography@phs.scot"
+      ),
+      call = NULL, rlang_backtrace_on_error = "none"
+    )
+  }
+
   data <- switch(ext,
     "rds" = readr::read_rds(file = path),
```

Comment:
I tried `get_spd(version = "2024_1")` and it returned a (non-tibble) dataframe, I think we need to add the explicit conversion, so that our function always returns a tibble.
```suggestion
    "rds" = tibble::as_tibble(readr::read_rds(file = path)),
```